### PR TITLE
[Refactor] refactor flinkClient

### LINKED ISSUE #2339: [Refactor] Refactor FlinkClientHandler
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

# what
- Responsibility： The FlinkClientHandler is responsible for Flink client action, such as ,deploy, submit.... 
- The every method contains muti-case lines style， which makes it compatible with multiple execution mode

# why
A new method is added each time, it needs to follow the existing coding.
There is a certain amount of code bloat as methods are introduced as discussed in https://github.com/apache/incubator-streampark/pull/2268#discussion_r1103619008
# how
Wait for discussion.

### Usage Scenario

N.A

### Related issues

https://github.com/apache/incubator-streampark/pull/2268

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by wolfboys**
[Refactor] refactor flinkClient

Issue Number: close #2339 

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes / no)

### CODE REVIEW

**Comment by wolfboys** [comment]
cc @1996fanrui  @RocMarshal PTAL, thx

File: streampark-flink/streampark-flink-client/streampark-flink-client-api/src/main/scala/org/apache/streampark/flink/client/FlinkClient.scala [hunk 1]
```diff
@@ -17,144 +17,59 @@
...
+  private[this] val DEPLOY_REQUEST = "org.apache.streampark.flink.client.bean.DeployRequest" -> "deploy"
 
-  private[this] val CANCEL_REQUEST_CLASS_NAME = "org.apache.streampark.flink.client.bean.CancelRequest"
+  private[this] val CANCEL_REQUEST = "org.apache.streampark.flink.client.bean.CancelRequest" -> "cancel"
 
-  private[this] val SHUTDOWN_REQUEST_CLASS_NAME = "org.apache.streampark.flink.client.bean.ShutDownRequest"
+  private[this] val SHUTDOWN_REQUEST = "org.apache.streampark.flink.client.bean.ShutDownRequest" -> "shutdown"
 
-  private[this] val SAVEPOINT_REQUEST_CLASS_NAME = "org.apache.streampark.flink.client.bean.TriggerSavepointRequest"
+  private[this] val SAVEPOINT_REQUEST = "org.apache.streampark.flink.client.bean.TriggerSavepointRequest" -> "triggerSavepoint"
```
Comment (RocMarshal) [inline streampark-flink/streampark-flink-client/streampark-flink-client-api/src/main/scala/org/apache/streampark/flink/client/FlinkClient.scala:40]: What about making the mapping be abstracted as a behavior and  fall the mapping info into the concrete  Request implementation?

File: streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/FlinkClientHandler.scala [hunk 2]
```diff
@@ -25,55 +25,55 @@ object FlinkClientHandler {
...
-      case ExecutionMode.KUBERNETES_NATIVE_SESSION => KubernetesNativeSessionSubmit.submit(submitInfo)
-      case ExecutionMode.KUBERNETES_NATIVE_APPLICATION => KubernetesNativeApplicationSubmit.submit(submitInfo)
+      case ExecutionMode.LOCAL => LocalClient.submit(submitInfo)
+      case ExecutionMode.REMOTE => RemoteClient.submit(submitInfo)
+      case ExecutionMode.YARN_APPLICATION => YarnApplicationClient.submit(submitInfo)
+      case ExecutionMode.YARN_SESSION => YarnSessionClient.submit(submitInfo)
+      case ExecutionMode.YARN_PER_JOB => YarnPerJobClient.submit(submitInfo)
+      case ExecutionMode.KUBERNETES_NATIVE_SESSION => KubernetesNativeSessionClient.submit(submitInfo)
+      case ExecutionMode.KUBERNETES_NATIVE_APPLICATION => KubernetesNativeApplicationClient.submit(submitInfo)
       case _ => throw new UnsupportedOperationException(s"Unsupported ${submitInfo.executionMode} submit ")
```
Comment (RocMarshal) [inline streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/FlinkClientHandler.scala]: There's some determinate condition in the client.
1. The ExecutionMode field is required for every existed XXXRequest and behavior like these lines.
2. every request need a method to drive the action bounded with `xxxClient`
3. special ExecutionMode with special xxxClient.

If we want to remove the case lines , which was mentioned in the `trigger savepoint PR`, here's 2 optional raw strategy from me:

1. hold a map to store the `ExecutionMode -> Concrete FlinkClientTrait`, then use mapping to judge the `Concrete FlinkClientTrait` to drive corresponding method. 
In this way, if we introduce a new client action, we need   
-  add a `xxxRequest` first,
-  add a  xxxrequest->method name  mapping.
-  then, add a method declaration in `FlinkClientTrait` , 
- next,  add the corresponding method implementations in `xxxClient` with its' executionMode,  
- finally, add the same method for `FlinkClientHandler` ()
In other words, we only reduce the redundant case lines in every method of `FlinkClientHandler` by this way. but can't follow the OCP rule well.

2.  Introduce  a abstract base class that has `identifier` behavior to represent the method name and a `getFlinkClientTrait`  behavior to represent the mapping from `execution mode` to `Concrete FlinkClientTrait`.
And make `FlinkClientHandler` as a child class of `FlinkClientTrait`,which is helpful to unify the interface about Client classes. 
In this way, if we introduce a new client action, we need   
-  add a `xxxRequest` first, we need add the required implementations from the above abstract parent, 
-  then, add a method declaration in `FlinkClientTrait` , 
- next,  add the corresponding method implementations in all of children object difinetion.

We can reduce the redundant case lines in every method of `FlinkClientHandler` by this way and follow the OCP rule better. However, the `method name` attribute of xxxRequest used to drive in the xxxClient is a little weird for me.

If I'm wrong, correct me, please. Any suggestion is appreciated.
Thx ~

**Review by RocMarshal — COMMENTED** [review]
Hi, @wolfboys Thanks for the contribution. 
I left some raw comments. PTAL in your free time.

CC @1996fanrui

File: streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/FlinkClientHandler.scala (hunk 2 above)
Comment (wolfboys) [reply streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/FlinkClientHandler.scala]: > There's some determinate condition in the client.
> 
> 1. The ExecutionMode field is required for every existed XXXRequest and behavior like these lines.
> 2. every request need a method to drive the action bounded with `xxxClient`
> 3. special ExecutionMode with special xxxClient.
> 
> If we want to remove the case lines , which was mentioned in the `trigger savepoint PR`, here's 2 optional raw strategy from me:
> 
> 1. hold a map to store the `ExecutionMode -> Concrete FlinkClientTrait`, then use mapping to judge the `Concrete FlinkClientTrait` to drive corresponding method.
>    In this way, if we introduce a new client action, we need
> 
> * add a `xxxRequest` first,
> * add a  xxxrequest->method name  mapping.
> * then, add a method declaration in `FlinkClientTrait` ,
> * next,  add the corresponding method implementations in `xxxClient` with its' executionMode,
> * finally, add the same method for `FlinkClientHandler` ()
>   In other words, we only reduce the redundant case lines in every method of `FlinkClientHandler` by this way. but can't follow the OCP rule well.
> 
> 2. Introduce  a abstract base class that has `identifier` behavior to represent the method name and a `getFlinkClientTrait`  behavior to represent the mapping from `execution mode` to `Concrete FlinkClientTrait`.
>    And make `FlinkClientHandler` as a child class of `FlinkClientTrait`,which is helpful to unify the interface about Client classes.
>    In this way, if we introduce a new client action, we need
> 
> * add a `xxxRequest` first, we need add the required implementations from the above abstract parent,
> * then, add a method declaration in `FlinkClientTrait` ,
> * next,  add the corresponding method implementations in all of children object difinetion.
> 
> We can reduce the redundant case lines in every method of `FlinkClientHandler` by this way and follow the OCP rule better. However, the `method name` attribute of xxxRequest used to drive in the xxxClient is a little weird for me.
> 
> If I'm wrong, correct me, please. Any suggestion is appreciated. Thx ~

Thanks for your feedback and comments, updated .

File: streampark-flink/streampark-flink-client/streampark-flink-client-api/src/main/scala/org/apache/streampark/flink/client/FlinkClient.scala (hunk 1 above)
Comment (wolfboys) [reply streampark-flink/streampark-flink-client/streampark-flink-client-api/src/main/scala/org/apache/streampark/flink/client/FlinkClient.scala:40]: > What about making the mapping be abstracted as a behavior and fall the mapping info into the concrete Request implementation?

Both ways are fine. In terms of code readability and cleanliness, I prefer the current code style.

**Review by muchunjin — APPROVED** [review]
LGTM

**Comment by RocMarshal** [comment]
thanks @wolfboys for the update.
LGTM no-binding.

**Pull request merged**